patroni/dcs/consul.py

```diff
@@ -232,6 +232,9 @@ def create_session(self):
                 logger.info('waiting on consul')
                 time.sleep(5)
 
+    def reload_config(self, config):
+        self.token = config['token']
```

Comment:
There is no `token` property in the `Consul` object, it is defined and used in the [ConsulClient](https://github.com/zalando/patroni/blob/master/patroni/dcs/consul.py#L125-L135) and [HTTPClient](https://github.com/zalando/patroni/blob/master/patroni/dcs/consul.py#L41).
Basically you have to implement `set_token` method in the `ConsulClient`, which will set the new value to `self._token` and take care about updating the token value in the existing `HTTPClient` object.

Reply:
Yep, stupid oversight sorry. I've done this and updated the PR

Reply:
Should be done